netkan/netkan/webhooks/spacedock_add.py

```diff
@@ -43,8 +44,21 @@ def add_hook(game_id: str) -> Tuple[str, int]:
     return '', 204
 
 
-def batch_message(raw: Dict[str, Any], game_id: str) -> SendMessageBatchRequestEntryTypeDef:
-    body = json.dumps(raw)
+def batch_message(raw: 'ImmutableMultiDict[str, str]', game_id: str) -> SendMessageBatchRequestEntryTypeDef:
+    body = json.dumps({**raw,
+                       # Turn the separate user property lists into a list of user dicts so JSON can encode it
+                       # (the original properties will only have the first user)
+                       'all_authors': [{'username':            user_tuple[0],
+                                        'user_github':         user_tuple[1],
+                                        'user_forum_id':       user_tuple[2],
+                                        'user_forum_username': user_tuple[3],
+                                        'email':               user_tuple[4]}
+                                       for user_tuple
+                                       in zip(raw.getlist('username'),
+                                              raw.getlist('user_github'),
+                                              raw.getlist('user_forum_id'),
+                                              raw.getlist('user_forum_username'),
+                                              raw.getlist('email'))]})
```

Comment:
Are all these keys going to be present if there are multiple authors?

Reply:
That's the plan, yeah. They're already sent unconditionally, regardless of if there is one author or twelve, and KSP-SpaceDock/SpaceDock#490 is just making them contain lists:

https://github.com/KSP-SpaceDock/SpaceDock/pull/490/files

Reply:
I've crafted a test for this, intended destination `tests.webhooks.TestWebhookSpaceDockAdd`

```python
    @mock.patch('netkan.webhooks.config.WebhooksConfig.client')
    def test_multi_authors_forms(self, queued: MagicMock):
        data = self.mock_netkan_hook()
        data.update({
            'username': ['author1', 'author2'],
            'user_github': ['author1_gh']
        })
        response = self.client.post('/sd/add/ksp', data=data)
        self.assertEqual(response.status_code, 204)
        call = queued.method_calls.pop().call_list().pop()
        body = json.loads(call[2].get('Entries')[0].get('MessageBody'))
        self.assertListEqual(
            body.get('all_authors'),
            [{'username': 'author1', 'user_github': 'author1_gh',
                'email': 'modauthor1@gmail.com'}, {'username': 'author2'}]
        )
```

Which fails:
```
AssertionError: Lists differ: [] != [{'username': 'author1', 'user_github': 'author1_gh'}, {'username': 'author2'}]

Second list contains 2 additional elements.
First extra element 0:
{'username': 'author1', 'user_github': 'author1_gh'}

- []
+ [{'user_github': 'author1_gh', 'username': 'author1'}, {'username': 'author2'}]
```

The problem is the zip, it'll return a tuple with a length equal to the item with the *least* items. Which will be fine if the lists come through in equal lengths, regardless of if an author is missing  a github or something.

If the payload comes through more like this, the lists will all be equal
```
('user', 'author1'), ('user', 'author2'),('user_github', 'author3')
('user_github', 'author1_gh'), ('user_github', ''),('user_github', 'author3_gh')
('email', 'author1@email'), ('email', ''),('email': '')
('user_forum_id', ''), ('user_forum_id', ''),(user_forum_id': '')
('user_forum_username', ''), ('user_forum_username', ''),('user_forum_username': '')
```